gateway-shell/src/main/java/org/apache/knox/gateway/shell/KnoxSh.java

```diff
@@ -155,6 +171,84 @@ public boolean validate() {
     public abstract String getUsage();
   }
 
+  private class KnoxBuildTrustStore extends Command {
+
+    private static final String USAGE = "buildTrustStore --gateway server-url";
+    private static final String DESC = "Downloads the gateway server's public certificate and builds a trust store.";
+    private static final String GATEWAY_CERT_NOT_EXPORTED = "Finished work without building truststore";
+    private static final String GATEWAY_CERT_EXPORTED_MESSAGE_PREFIX = "Gateway server's certificate is exported into ";
+
+    @Override
+    public void execute() throws Exception {
+      String result = GATEWAY_CERT_NOT_EXPORTED;
+      try {
+        final X509Certificate gatewayServerPublicCert = fetchPublicCertFromGatewayServer();
+        if (gatewayServerPublicCert != null) {
+          final File trustStoreFile = Paths.get(System.getProperty("user.home"), KnoxSession.GATEWAY_CLIENT_TRUST).toFile();
```

Comment:
I know there are a few notes about this but wanted to call out that we should make this match with existing behavior with the env variables.